extensions/vault/runtime/src/main/java/io/quarkus/vault/VaultServerRuntimeConfig.java

```diff
@@ -0,0 +1,132 @@
+package io.quarkus.vault;
+
+import static io.quarkus.vault.LogConfidentialityLevel.LOW;
+
+import java.time.Duration;
+import java.util.Optional;
+
+import io.quarkus.runtime.annotations.ConfigItem;
+import io.quarkus.runtime.annotations.ConfigPhase;
+import io.quarkus.runtime.annotations.ConfigRoot;
+
+@ConfigRoot(name = "vault", phase = ConfigPhase.RUN_TIME)
+public class VaultServerRuntimeConfig {
+
+    public static final String DEFAULT_K8S_TOKEN_PATH = "/var/run/secrets/kubernetes.io/serviceaccount/token";
+    public static final String DEFAULT_KV_SECRET_ENGINE_PATH = "secret";
+    public static final String KV_SECRET_ENGINE_VERSION_V1 = "1";
+    public static final String KV_SECRET_ENGINE_VERSION_V2 = "2";
+    public static final String DEFAULT_RENEW_GRACE_PERIOD = "1H";
+    public static final String DEFAULT_CACHE_PERIOD = "10M";
+
+    /**
+     * vault url. eg https://localhost:8200
+     */
+    @ConfigItem
+    public Optional<String> url;
+
+    /**
+     * k8s auth role
+     */
+    @ConfigItem
+    public Optional<String> role;
+
+    /**
+     * authentication type when logging in to get a client token. possible values:
+     * - k8s: kubernetes authentication as defined in https://www.vaultproject.io/api/auth/kubernetes/index.html
+     */
+    @ConfigItem(defaultValue = "K8S")
```

Comment:
This type assumes a more involved deployment case. Should a user pass be a default value ?

Reply:
initially I was supporting k8s only, which seemed fine with quarkus own objectives (targeting specifically k8s).
I only added userpass for testing purposes.
the issue I have with userpass is that you have to put a password in your configuration, which provides access to vault where you have your secrets. so it is kind of awkard to put all of these efforts to secure your secrets in vault, if you are juste going to provide the credentials in clear in your config.
that is why it seemed more natural to have in default the platform we are targeting first.

Reply:
+1